src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1520,4 +1520,44 @@ public static void SetTestHook(string property, object value)
             }
         }
     }
+
+    /// <summary>
+    /// An bounded stack based on a linked list.
+    /// </summary>
+    internal class BoundedStack<T> : LinkedList<T>
+    {
+        private readonly int _capacity;
+
+        /// <summary>
+        /// Lazy initialisation, i.e. it sets only its limit but does not allocate the memory for the given capacity.
+        /// </summary>
+        /// <param name="capacity"></param>
+        internal BoundedStack(int capacity)
+        {
+            _capacity = capacity;
+        }
+
+        /// <summary>
+        /// Push item.
+        /// </summary>
+        /// <param name="item"></param>
+        internal void Push(T item)
+        {
+            this.AddFirst(item);
+
+            if(this.Count > _capacity)
```

Comment:
Maybe include braces for the `if` and drop the `this` from property references. That seems to be the style convention of the file.

Reply:
I agree, normally, it is me who comments on that on other people's code.
![Alt Text](https://media.giphy.com/media/kYkQYXkO3XyRa/giphy.gif)

Reply:
Yep, I agree. Fixed.